plugins/drivers/base/plugin_test.go

```diff
@@ -0,0 +1,125 @@
+package base
+
+import (
+	"bytes"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/nomad/plugins/drivers/base/proto"
+	"github.com/stretchr/testify/require"
+	"github.com/ugorji/go/codec"
+	"golang.org/x/net/context"
+)
+
+type testDriverState struct {
+	Pid int
+	Log string
+}
+
+func TestBaseDriver_RecoverTask(t *testing.T) {
```

Comment:
Hmm why not use the harness so you test the entire client/server/proto conversion code path?

Reply:
I think these tests were written before I created the harness. Plus these tests are handling specifically the logic that translates from the protobuf api/structs to the go ones.